lib/features/route_map/widgets/bottom_sheet/tiles/route_tile.dart

```diff
@@ -0,0 +1,113 @@
+import "route_stat.dart";
+
+class RouteTile extends ConsumerStatefulWidget {
+  const RouteTile({super.key, required this.route});
+
+  final Route route;
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => RouteTileState();
+}
+
+class RouteTileState extends ConsumerState<RouteTile> {
+  late RouteDetailsOption _chosenOption;
+
+  @override
+  void initState() {
+    super.initState();
+    _chosenOption = RouteDetailsOption.info;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ExpansionTile(
+      title: Row(
+        children: [
+          Expanded(
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Text(widget.route.name, style: context.textTheme.headlineSmall?.copyWith(fontSize: 22)),
+                const SizedBox(height: 6),
+                Text(widget.route.distance.inKilometers()),
+              ],
+            ),
+          ),
+          RouteStat(
+            icon: Assets.icons.water.svg(width: SelectRouteConfig.iconSize, height: SelectRouteConfig.iconSize),
+            comment: widget.route.requiredWater?.inMililiters() ?? context.l10n.ml_0,
+          ),
+          RouteStat(
+            icon: Assets.icons.time.svg(width: SelectRouteConfig.iconSize, height: SelectRouteConfig.iconSize),
+            comment: widget.route.estimatedTime.inMinutes(),
+          ),
+          RouteStat(
+            icon: Assets.icons.kcal.svg(width: SelectRouteConfig.iconSize, height: SelectRouteConfig.iconSize),
+            comment: widget.route.calories?.inKcal() ?? context.l10n.kcal_0,
+          ),
+        ],
+      ),
+      children: [
+        Padding(
+          padding: const EdgeInsets.all(AppPaddings.tinySmall),
+          child: SecondaryActionButton(
+            onPressed: () async {
+              await showDialog<StartRouteModal>(
+                context: context,
+                builder: (context) => StartRouteModal(route: widget.route),
+              );
+            },
+            text: context.l10n.start_route,
+          ),
+        ),
+        Padding(
+          padding: const EdgeInsets.all(AppPaddings.tinySmall),
+          child: Row(
+            spacing: BottomSheetHeaderConfig.controlsSpacing,
+            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+            children: [
+              SecondaryActionButton(
+                onPressed: () {
+                  setState(() => _chosenOption = RouteDetailsOption.info);
+                },
+                text: context.l10n.route_description,
+                backgroundColor:
+                    _chosenOption != RouteDetailsOption.info ? ColorConsts.whiteGray : ColorConsts.lightGreen,
+                textColor: _chosenOption != RouteDetailsOption.info ? ColorConsts.lightGreen : ColorConsts.whiteGray,
+              ),
+
+              SecondaryActionButton(
+                onPressed: () {
```

Comment:
pressing these two buttons causes exceptions, Incorrect use of ParentDataWidget.